### ISSUE TITLE
Proxy server and environment variable support

### ISSUE THREAD

**Issue by GeoffWilliams** [puppetlabs/puppetlabs-aws#154]
# Summary
Module doesn't work with proxy servers due to environment variables and settings from `puppet.conf` not being parsed

# Problems this causes
* Module unusable in environments requiring a proxy server (unless the ruby code is edited)
* There are many other options to the AWS client that users may want or need to set http://docs.aws.amazon.com/sdkforruby/api/Aws/CloudSearchDomain/Client.html and this is not possible at the moment

# Workaround
File:  https://github.com/puppetlabs/puppetlabs-aws/blob/1.0.0/lib/puppet_x/puppetlabs/aws.rb#L55
Customer found that he was able to gain proxy support by changing the initialisation of the EC2 client:

*Old line*
```
::Aws::EC2::Client.new({region: region})
```
*New line*
```
::Aws::EC2::Client.new({region: region, http_proxy: "https://internal-proxy.account:8080/"})
```

# Long-term solution
Ideally the initialiser should read standard environment variables (http(s)_proxy, etc) or parse proxy details from `puppet.conf`.

Further variables could be parsed from command line variables, Or the AWS config file

# Problem scope
This fix needs to be applied anywhere a new AWS client instance is created.  Customer has identified at least the following calls:
```
::Aws::ElasticLoadBalancing::Client.new({region: region})
::Aws::AutoScaling::Client.new({region: region})
::Aws::CloudWatch::Client.new({region: region})
::Aws::Route53::Client.new({region: region})
```

**Comment by aharden** [comment]
+1.  The EC2 workaround enabled me to provision EC2 resources in our AWS VPC.

**Comment by garethr** [comment]
Thanks @GeoffWilliams for the details. Just a note that this is up for internal review at https://github.com/puppetlabs/puppetlabs-aws/pull/163 and will be in the next release of the module.

**Comment by garethr** [comment]
This has been merged and is also available in 1.1. Thanks for the detailed bug report.
